### ISSUE TITLE
How to choose a trained model

### ISSUE THREAD

**Issue by czc567** [Y-debug-sys/Diffusion-TS#16]
Hello, there is no validation set when doing unconditional generation, so I am confused about how to select the model from training as the final model.
Will the final report results be uniformly based on the results when milestone=10? Or you evaluate the models under different milestones and choose the best one.

**Comment by Y-debug-sys** [comment]
Following TimeGAN, we did not use validation set, which may need to possible overfitting, but also made the total number of training epochs a parameter that needs optimization. I usually test trained models sequentially from 10 to smaller milestone, then change the number of training epochs. After that, I would determinate the number of epochs, then take milestone-10 as the final model. Of course, training for a few more epochs to select the best model from different milestones is also acceptable.

**Comment by czc567** [comment]
Thank you for your reply. I also want to ask a question. The results reported in the paper are after sampling a batch, then evaluating the same batch of synthetic data 5 times and averaging. Or sample 5 batches, evaluate each batch separately and then average.
I've found that the former results in slightly different results each time.
Note: The batch here represents the same number as the training dataset.

**Comment by Y-debug-sys** [comment]
Hi, are you referring to metrics implemented in PyTorch? We use the latter, evaluating all baselines under the same seed due to its randomness.  Regarding discriminative score and predictive score, we just repeat experiments in the original TimeGAN paper. Thanks.

**Comment by czc567** [comment]
Yes. The Context_FID result of your paper is to run metric_pytorch.ipynb once after sampling the sample. Is my understanding correct?

**Comment by Y-debug-sys** [comment]
Yes, your understanding is OK.

**Comment by czc567** [comment]
Thanks, you solved my doubts.
